sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -401,7 +401,9 @@ class SessionCatalog(
       val db = formatDatabaseName(name.database.getOrElse(currentDb))
       val table = formatTableName(name.table)
       val relation =
-        if (name.database.isDefined || !tempTables.contains(table)) {
+        if (db == "information_schema") {
+          tempTables(s"$db.$table")
```

Comment:
That's a good idea!

I made `SessionCatalog` object in the following PR to store 'DEFAULT'.

https://github.com/apache/spark/pull/14115/files#diff-b3f9800839b9b9a1df9da9cbfc01adf8R38

We can move `INFORMATION_SCHEMA` to there!
Thanks.